Tests/MongoSwiftTests/SpecTestRunner/SpecTestUtils.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+@testable import MongoSwift
+import XCTest
+
+extension MongoSwiftTestCase {
+    /// Gets the path of the directory containing spec files, depending on whether
+    /// we're running from XCode or the command line
+    static var specsPath: String {
+        // if we can access the "/Tests" directory, assume we're running from command line
+        if FileManager.default.fileExists(atPath: "./Tests") {
+            return "./Tests/Specs"
+        }
+        // otherwise we're in Xcode, get the bundle's resource path
+        guard let path = Bundle(for: self).resourcePath else {
+            XCTFail("Missing resource path")
+            return ""
+        }
+        return path
+    }
+}
+
+extension Document {
+    init(fromJSONFile file: URL) throws {
+        let jsonString = try String(contentsOf: file, encoding: .utf8)
+        try self.init(fromJSON: jsonString)
+    }
+}
+
+/// Given a spec folder name (e.g. "crud") and optionally a subdirectory name for a folder (e.g. "read") retrieves an
+/// array of [(filename, file decoded to type T)].
+internal func retrieveSpecTestFiles<T: Decodable>(specName: String, 
+                                                subdirectory: String? = nil,
+                                                asType: T.Type) throws -> [(String, T)] {
```

Comment:
this tuple approach is much better than what I was doing before with setting the name after initialize the file...